updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "base64"
+require "dependabot/file_parsers"
+
+# This class describes the dependencies obtained from a project at a specific commit SHA
+# including both the Dependabot::DependencyFile objects at that reference as well as
+# means to parse them into a set of Dependabot::Dependency objects.
+#
+# This class is the input for a Dependabot::Updater process with Dependabot::DependencyChange
+# representing the output.
+module Dependabot
+  class DependencySnapshot
```

Comment:
First of all, I really dig the clear lines that are being drawn in the updater 🎉 I think these recent changes just make sense 😄 

💅 I wonder if we can come up with a stronger signal for this being an input. The `DependencyChange` name suggests that its responses will represent output, but the `DependencySnapshot` name doesn't resonate _with me_ as an input per se. Maybe that's the way it should be, and the names of variables should be holding meaning.

No need to update this, but if that resonates with you, I'm happy to try to work with you to come up with a name that reflects the before state of the dependency file 🙂

Reply:
Yeah, I'm not completely happy with this name but I also don't think the responsibilities of this class are fully final so I'd prefer to revisit this after a few more transforms once the parsing is properly contained